src/scp/test/QuorumIntersectionTests.cpp

```diff
@@ -0,0 +1,700 @@
+// Copyright 2016 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "crypto/Hex.h"
+#include "crypto/SHA.h"
+#include "lib/catch.hpp"
+#include "main/Config.h"
+#include "scp/QuorumIntersectionChecker.h"
+#include "test/test.h"
+#include "util/Logging.h"
+#include "util/Math.h"
+#include "xdrpp/marshal.h"
+#include <lib/util/format.h>
+#include <xdrpp/autocheck.h>
+
+using namespace stellar;
+
+using QS = SCPQuorumSet;
+using VQ = xdr::xvector<QS>;
+using VK = xdr::xvector<PublicKey>;
+using std::make_shared;
+
+TEST_CASE("quorum intersection basic 4-node", "[scp][quorumintersection]")
+{
+    QuorumTracker::QuorumMap qm;
+
+    PublicKey pkA = SecretKey::pseudoRandomForTesting().getPublicKey();
+    PublicKey pkB = SecretKey::pseudoRandomForTesting().getPublicKey();
+    PublicKey pkC = SecretKey::pseudoRandomForTesting().getPublicKey();
+    PublicKey pkD = SecretKey::pseudoRandomForTesting().getPublicKey();
+
+    qm[pkA] = make_shared<QS>(2, VK({pkB, pkC, pkD}), VQ{});
+    qm[pkB] = make_shared<QS>(2, VK({pkA, pkC, pkD}), VQ{});
+    qm[pkC] = make_shared<QS>(2, VK({pkA, pkB, pkD}), VQ{});
+    qm[pkD] = make_shared<QS>(2, VK({pkA, pkB, pkC}), VQ{});
+
+    Config cfg(getTestConfig(0, Config::TESTDB_IN_MEMORY_SQLITE));
```

Comment:
`getTestConfig` has proper defaults, so arguments can be avoided

Reply:
fixed.